mailbox/postgres/src/test/java/org/apache/james/mailbox/postgres/PostgresMailboxManagerTest.java

```diff
@@ -179,7 +178,7 @@ void deleteMessageInMailboxShouldNotDeleteReferencedMessageMetadata() throws Exc
 
         @Test
         void deleteMailboxShouldEventuallyDeleteUnreferencedMessageMetadataWhenDeletingMailboxMessageFail() throws Exception {
-            PostgresMailboxMessageDAO spiedPostgresMailboxMessageDAO = spyPostgresMailboxMessageDAO();
+            PostgresMailboxMessageDAO spiedPostgresMailboxMessageDAO = spy(PostgresMailboxManagerProvider.postgresMailboxMessageDAO);
```

Comment:
I think @chibenwa said to do that in the constructor... You have a PostgresMessageDAO and a PostgresMailboxMessageDAO declared in the setup, but here you create new local ones. It's not necessary.

Just do directly:

```
@BeforeEach
        void setUp() throws Exception {
            
            [...]

            postgresMessageDAO = spy(new PostgresMessageDAO(postgresExtension.getPostgresExecutor(), new HashBlobId.Factory()));
            postgresMailboxMessageDAO = spy(new PostgresMailboxMessageDAO(postgresExtension.getPostgresExecutor()));
        }
```
Then use those in your tests without the need to redeclare local objects?

Reply:
postgresMessageDAO used in test method and postgresMessageDAO used in mailboxManager are different instances

Reply:
Alright, missed that. Ok then :)